plugins/GeoIp2/LocationProvider/GeoIp2/ServerModule.php

```diff
@@ -327,10 +327,13 @@ protected static function getSystemSettingsValues()
 
         $settingValues = self::$defaultGeoIpServerVars; // preset with defaults
 
-        $systemSettings = new SystemSettings();
+        try {
+            $systemSettings = new SystemSettings();
 
-        foreach ($systemSettings->geoIp2variables as $name => $setting) {
-            $settingValues[$name] = $setting->getValue();
+            foreach ($systemSettings->geoIp2variables as $name => $setting) {
+                $settingValues[$name] = $setting->getValue();
+            }
+        } catch (\Exception $e) {
         }
 
         $cache->save($cacheKey, $settingValues);
```

Comment:
Is it ok to save these values in the cache if getting the system settings fails?

Reply:
that should only occur in tests. But guess it's better not to cache on failure